zebra-state/src/service.rs

```diff
@@ -468,13 +469,6 @@ impl StateService {
         let queued_prev_hash = finalized.block.header.previous_block_hash;
```

Comment:
This PR appears to have a bug that causes syncing to hang under some network conditions:
> 2023-03-22T04:55:46.396678Z  WARN {net="Main"}: zebrad::components::sync::progress: chain updates have stalled, state height has not increased for 160 minutes. Hint: check your network connection, and your computer clock and time zone sync_percent=99.993% current_height=Height(2024520) network_upgrade=Nu5 time_since_last_state_block=2h 40m target_block_spacing=PT75S max_block_spacing=None is_syncer_stopped=false

https://github.com/ZcashFoundation/zebra/actions/runs/4485179712/jobs/7887130253?pr=6335

Since this is supposed to be an optimisation-only PR, I'm concerned that we're making more changes that are actually needed. (It would be ok to re-download some redundant blocks, but it's not ok to skip the re-download of a block that's actually needed.)

 I'd suggest reverting any unnecessary changes, or splitting the PR into different stages.

Reply:
I'm wondering what happens when:
1. A block is sent to the channel and put in the queued list
2. That block is rejected by the block write task due to a validation error
3. But it stays in the queued list because the rejection doesn't change what's in the queued list

As an alternative, we could just accept that we might sometimes re-download queued blocks across the syncer and inbound service. (Each download service already checks the block hashes of its own pending tasks, and doesn't download duplicate blocks. Those tasks only return once the block has been committed successfully.)